performance_test.go

```diff
@@ -246,8 +251,80 @@ var _ = Describe("Authorizing requests", Serial, Ordered, func() {
 		}, gmeasure.SamplingConfig{N: 30, Duration: 2 * time.Minute})
 		// we'll sample the function up to 30 times or up to 2 minutes, whichever comes first.
 	})
+
+	It("efficiently synchronizes access cache", Serial, Label("perf"), func(ctx context.Context) {
+		// new gomega experiment
+		experiment := gmeasure.NewExperiment("Access Cache Synch")
+
+		// Register the experiment as a ReportEntry - this will cause Ginkgo's reporter infrastructure
+		// to print out the experiment's report and to include the experiment in any generated reports
+		AddReportEntry(experiment.Name, experiment)
+
+		// create resourceCache, namespacelister, and handler
+		resourceCache, err := BuildAndStartResourceCache(ctx, cacheCfg)
+		utilruntime.Must(err)
+		c, err := buildAndStartAccessCache(ctx, resourceCache)
+		utilruntime.Must(err)
+
+		// check cache is correctly populated with
+		// more than 5000 subjects
+		// and more than 10000 total namespaces
+		cacheData := unsafeGetPrivateCacheData(c.AccessCache)
+		Expect(len(cacheData)).To(BeNumerically(">", 5000))
+		Expect(cacheData).To(Satisfy(func(d cache.AccessData) bool {
```

Comment:
Shouldn't this closure operate on a `map[rbacv1.Subject][]corev1.Namespace`?

Reply:
This way if we make changes to what AccessCache uses, it will fail at compile time